### PR TITLE
Update TaskRun List Page Columns & PipelineRun Tab

### PR DESCRIPTION
**Pull request by debsmita1**
**JIRA story:**
https://issues.redhat.com/browse/ODC-4800

**Solution Description:**
- added `Task Runs` tab in the pipelineRun details page which lists all the Task Runs associated with the Pipeline Run
- Added columns `pipeline` , `task` & `pod`  & removed `duration`
- The pipeline column is not shown in the TaskRun Tab under Pipeline Run details page
- fixed the TaskRun list in the Search page
- added tests for `TaskRunsRow`

**GIF/Screenshots:**
PDtaskruns
taskruns1
TaskRuns
all-pro
search

**Test Coverage:**
test1

### CODE REVIEW

File: frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsHeader.tsx [hunk 1]
```diff
@@ -10,33 +10,39 @@ const TaskRunsHeader = () => {
       props: { className: tableColumnClasses[0] },
     },
     {
-      title: 'Namespace',
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsHeader.tsx:13]: Can you really remove Namespace? I think if you go to the Admin tab page and do all projects, you'll be missing this column, no?

Comment (debsmita1) [reply frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsHeader.tsx]: @andrewballantyne Thank you for pointing this out. I noticed in the pipelines list page that the namespace column shows up when the user is not in the context of a project. I have incorporated that in my PR and have added a GIF in PR description

File: frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsListPage.tsx [hunk 2]
```diff
@@ -2,29 +2,33 @@ import * as React from 'react';
...
 import { ListPage } from '@console/internal/components/factory';
 import { referenceForModel } from '@console/internal/module/k8s';
+import { getURLSearchParams } from '@console/internal/components/utils';
 import TaskRunsList from './TaskRunsList';
 import { TaskRunModel } from '../../../models';
 import { runFilters as taskRunFilters } from '../../pipelines/detail-page-tabs/PipelineRuns';
 
 interface TaskRunsListPageProps {
   hideBadge?: boolean;
+  showCreateButton?: boolean;
```
Comment (abhinandan13jan) [inline frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsListPage.tsx:12]: Is it passed by a parent component somewhere?

File: frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/TaskRuns.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import TaskRunsListPage from '../../taskruns/list-page/TaskRunsListPage';
+
+const TaskRuns: React.FC = (props) => (
+  <TaskRunsListPage
+    showTitle={false}
+    selector={{ 'tekton.dev/pipelineRun': _.get(props, 'obj.metadata.name') }}
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/TaskRuns.tsx:8]: we could replace the lodash _.get with optional chaining and remove the lodash import. why to import for this simple usage, WDYT?

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineRunDetailsPage.tsx [hunk 4]
```diff
@@ -24,6 +25,11 @@ const PipelineRunDetailsPage: React.FC<DetailsPageProps> = (props) => {
       pages={[
         navFactory.details(PipelineRunDetails),
         navFactory.editYaml(viewYamlComponent),
+        {
+          href: 'TaskRuns',
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/pipelineruns/PipelineRunDetailsPage.tsx:29]: Please change this to lower case hyphenated words in the url, so that we are using similar url segments in both the perspectives.
```suggestion
          href: 'task-runs',
```

**Review by karthikjeeyar — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/pipelineruns/PipelineRunDetailsPage.tsx (hunk 4 above)
Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/pipelineruns/PipelineRunDetailsPage.tsx:29]: Odd, the Pipeline Details Page has a "Runs" Tab for Pipeline Runs 😕 

Definitely agree we should `kebab-case` the route names though.

File: frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/TaskRuns.tsx (hunk 3 above)
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/TaskRuns.tsx]: Done

File: frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsListPage.tsx (hunk 2 above)
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsListPage.tsx]: removed it

**Comment by bgliwa01** [comment]
Is there a tooltip on the failed status showing the error?

**Comment by bgliwa01** [comment]
will there be a separate PR adding a task run details page?

**Comment by debsmita1** [comment]
> will there be a separate PR adding a task run details page?

@bgliwa01  yes this is the PR https://github.com/openshift/console/pull/6851

> Is there a tooltip on the failed status showing the error?

No, there isn't

**Comment by andrewballantyne** [comment]
> > Is there a tooltip on the failed status showing the error?
> 
> No, there isn't

@bgliwa01 I don't _think_ we have a convention for showing a tooltip on any status 🤔  Can you point me at a case where we do this?

**Comment by bgliwa01** [comment]
@andrewballantyne the only reason I'm asking is because it's in the ACs. I can't get a pipeline run to fail to validate if we have it or not currently.

**Comment by bgliwa01** [comment]
Just verified that we don't show error in tooltip...I will ask PM

**Comment by andrewballantyne** [comment]
> @andrewballantyne the only reason I'm asking is because it's in the ACs. I can't get a pipeline run to fail to validate if we have it or not currently.

Damn, good catch @bgliwa01! I did not at all see that when I was validating the AC. I guess I saw "status" and assumed "well that's pretty standard". I have asked a question on the Epic.

**Review by jerolimov — CHANGES_REQUESTED** [review]
@debsmita1 Cool, could we also change the default order of TaskRuns to the "started" date (newest first), similar to PipelineRuns?

Imo this makes sense in the overall search list as well in the PipelineRun > TaskRuns tab.

File: frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsListPage.tsx [hunk 5]
```diff
@@ -14,17 +15,19 @@ const TaskRunsListPage: React.FC<Omit<
...
-  TaskRunsListPageProps> = (props) => {
+  TaskRunsListPageProps> = ({ hideBadge, ...props }) => {
+  const searchParams = getURLSearchParams();
+  const kind = searchParams?.kind;
+
   return (
     <ListPage
       {...props}
-      canCreate={false}
+      canCreate={kind?.includes(referenceForModel(TaskRunModel)) ?? false}
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsListPage.tsx:25]: Curious, why this change?

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

**Comment by andrewballantyne** [comment]
> @debsmita1 could we also change the default order of TaskRuns to the "started" date (newest first), similar to PipelineRuns?

Definitely this should be done. Name is going to be rough being the default sort.

File: frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsRow.tsx [hunk 6]
```diff
@@ -22,17 +23,45 @@ const TaskRunsRow: RowFunction<TaskRunKind> = ({ obj, index, key, style }) => {
...
+            name={obj.metadata.labels['tekton.dev/pipeline']}
+            namespace={obj.metadata.namespace}
+          />
+        ) : (
+          '-'
+        )}
       </TableData>
       <TableData className={tableColumnClasses[3]}>
-        <Timestamp timestamp={obj?.status?.startTime} />
+        {obj.spec.taskRef?.name ? (
```
[the remark this answers is not in the document]
Comment (karthikjeeyar) [reply frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsRow.tsx:41]: I have expanded our `TaskRunKind` in this PR https://github.com/openshift/console/pull/6867

File: frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsRow.tsx [hunk 7]
```diff
@@ -3,14 +3,15 @@ import { TableRow, TableData, RowFunction } from '@console/internal/components/f
...
+import { TaskRunModel, PipelineModel, TaskModel } from '../../../models';
 import { tableColumnClasses } from './taskruns-table';
 import { Status } from '@console/shared';
 import { pipelineRunFilterReducer as taskRunFilterReducer } from '../../../utils/pipeline-filter-reducer';
-import { pipelineRunDuration as taskRunDuration } from '../../../utils/pipeline-utils';
 
 const TaskRunsRow: RowFunction<TaskRunKind> = ({ obj, index, key, style }) => {
   const taskRunsReference = referenceForModel(TaskRunModel);
+  const taskReference = referenceForModel(TaskModel);
+  const pipelineReference = referenceForModel(PipelineModel);
```
[the remark this answers is not in the document]
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsRow.tsx]: moved them oustide of the `TaskRunsRow`component

File: frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsListPage.tsx (hunk 5 above)
Comment (debsmita1) [reply frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsListPage.tsx:25]: this adds the `create task run` button in the search page when the Task Runs resource type is selected

**Review by bgliwa01 — APPROVED** [review]
confirmed with PM that tooltip on error will be taken out of ACs for this epic...lgtm!!

**Comment by karthikjeeyar** [comment]
Verified locally, works fine
/lgtm

**Review by bgliwa01 — APPROVED** [review]
lgtm

**Review by rohitkrai03 — APPROVED** [review]
/approve
